### PR TITLE
add context command in server for users customize jvm options

### PR DESCRIPTION
**Pull request by SummerSun**

### CODE REVIEW

File: src/Tomcat/TomcatModel.ts [hunk 1]
```diff
@@ -33,6 +35,54 @@ export class TomcatModel {
...
     }
 
+    public async updateVMOptions(serverName: string) : Promise<void> {
+        const server: TomcatServer = this.getTomcatServer(serverName);
+        const installPath: string = server.getInstallPath();
+        const catalinaBase: string = path.join(server.getStoragePath());
+        const bootStrap: string = path.join(installPath, 'bin', 'bootstrap.jar');
+        const tomcat: string = path.join(installPath, 'bin', 'tomcat-juli.jar');
+        let result: string[] = [
+            `${Constants.CLASS_PATH_KEY} ${bootStrap}${path.delimiter}${tomcat}`,
```
Comment (andxu) [inline src/Tomcat/TomcatModel.ts]: ${path.join(bootStrap,tomcat)} is better

File: src/Tomcat/TomcatModel.ts [hunk 2]
```diff
@@ -33,6 +35,54 @@ export class TomcatModel {
...
+            `${Constants.CATALINA_BASE_KEY}=${catalinaBase}`,
+            `${Constants.CATALINA_HOME_KEY}=${installPath}`
+        ];
+
+        if (!await fse.pathExists(server.vmOptionFile)) {
+            server.vmOptions = result.concat([Constants.BOOTSTRAP_FILE, '"$@"']);
+            return;
+        }
+        await new Promise((resolve: () => void): void => {
+            const lineReader: readline.ReadLine = readline.createInterface({
```
Comment (andxu) [inline src/Tomcat/TomcatModel.ts]: search readline.createInterface for duplicate codes

File: resources/jvm.options [hunk 3]
```diff
@@ -0,0 +1,101 @@
+## JVM configuration
```
Comment (andxu) [inline resources/jvm.options]: Give a doc about the original source of this file.

File: src/Constants.ts [hunk 4]
```diff
@@ -26,6 +26,22 @@ export const RESTART_CONFIG_ID: string = 'restart_when_http(s)_port_change';
 // tslint:disable-next-line:no-http-string
 export const LOCALHOST: string = 'http://localhost';
 
+export const VM_OPTION_FILE: string = 'jvm.options';
```
Comment (andxu) [inline src/Constants.ts]: JVM_OPTION_FILE

File: src/Tomcat/TomcatModel.ts [hunk 5]
```diff
@@ -33,6 +35,54 @@ export class TomcatModel {
...
+                    });
+                    result = result.concat(line);
+                }
+            });
+            lineReader.on('close', () => {
+                let isTempDirConfigured: boolean = false;
+                result.forEach((r: string) => {
+                    if (r.indexOf(Constants.JAVA_IO_TEMP_DIR_KEY) >= 0) {
+                        isTempDirConfigured = true;
+                        return;
```
Comment (andxu) [inline src/Tomcat/TomcatModel.ts]: no need for return, you possibly need break here, so forEach is not suitable here, using find instead? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

Comment (SummerSun) [reply src/Tomcat/TomcatModel.ts]: find looks like a good choice

File: src/Tomcat/TomcatModel.ts (hunk 2 above)
Comment (SummerSun) [reply src/Tomcat/TomcatModel.ts]: The other two listener are different scenarios. 

One is watch the server.xml for server configuration changes.
The other is child_process stdout.

File: resources/jvm.options (hunk 3 above)
Comment (SummerSun) [reply resources/jvm.options]: OK, I didn't find a sample jvm.options from official sites and took the elastic search sample.

Comment (SummerSun) [reply resources/jvm.options]: OK, I didn't find a sample jvm.options from official sites and took the elastic search sample.

**Comment by SummerSun** [comment]
Reference the customizing jvm options behaviors in IntelliJ. Provide a simple jvm.options and warning users to edit it with caution.

**Pull request merged**